src/query.rs

```diff
@@ -68,9 +68,8 @@ pub fn parse(statement: &str) -> Result<ParseResult> {
 pub fn deparse(protobuf: &protobuf::ParseResult) -> Result<String> {
     let buffer = protobuf.encode_to_vec();
     let len = buffer.len() as size_t;
-    let input = unsafe { CStr::from_bytes_with_nul_unchecked(&buffer) };
-    let data = input.as_ptr() as *mut c_char;
-    let protobuf = PgQueryProtobuf { data: data, len: len };
+    let data = buffer.as_ptr() as *const c_char as *mut c_char;
```

Comment:
What does this double-cast do?

Reply:
First we cast from `*const u8` to `*const c_char`, as `c_char` may be a different type, like `i8` on some platforms. Then, the second cast is just ceremony as the C API requires `*mut c_char`. The pointer is never used to mutate the input so this is superfluous.

We can't directly cast `*const u8` to `*mut c_char` (as `c_char` may be `i8`), so we have to first change the type, then the mutability (or vice versa).